include/util.h

```diff
@@ -1,55 +1,170 @@
+ */
+#define UINT8_MAX_DIGITS 3
+
+/**
+ * @brief Avoid overflow when adding two u32 integers
+ *
+ * @param a left operator **a + b**
+ * @param b left operator **a + b**
+ *
+ * @return the result of **a + b** or **UINT32_MAX**
+ */
+uint32_t u32_protected_add(uint32_t a, uint32_t b);
+
+/**
+ * @brief Avoid overflow when adding two u16 integers
+ *
+ * @param a left operator **a + b**
+ * @param b left operator **a + b**
+ *
+ * @return the result of **a + b** or **UINT16_MAX**
+ */
+uint16_t u16_protected_add(uint16_t a, uint16_t b);
+
+/**
+ * @brief Avoid overflow when multiplying two u32 integers
+ *
+ * @param a left operator **a * b**
+ * @param b left operator **a * b**
+ *
+ * @return the result of **a * b** or **UINT32_MAX**
+ */
+uint32_t u32_protected_mult(uint32_t a, uint32_t b);
+
+/**
+ * @brief Avoid overflow when multiplying two u16 integers
+ *
+ * @param a left operator **a * b**
+ * @param b left operator **a * b**
+ *
+ * @return the result of **a * b** or **UINT16_MAX**
+ */
+uint16_t u16_protected_mult(uint16_t a, uint16_t b);
+
+/**
+ * @brief Get the number of digits in a number
+ * https://stackoverflow.com/questions/1068849/how-do-i-determine-the-number-of-digits-of-an-integer-in-c
+ *
+ * @param n value to find the number of decimal digits of
+ *
+ * @return the number of digits in a number
+ */
+static inline int get_digits(int n)
 {
-    if (n < 10) return 1;
-    if (n < 100) return 2;
-    if (n < 1000) return 3;
-    if (n < 10000) return 4;
-    if (n < 100000) return 5;
-    if (n < 1000000) return 6;
-    if (n < 10000000) return 7;
-    if (n < 100000000) return 8;
-    if (n < 1000000000) return 9;
+    if (n < 10)
+        return 1;
+    if (n < 100)
+        return 2;
+    if (n < 1000)
+        return 3;
+    if (n < 10000)
+        return 4;
+    if (n < 100000)
+        return 5;
+    if (n < 1000000)
+        return 6;
+    if (n < 10000000)
+        return 7;
+    if (n < 100000000)
+        return 8;
+    if (n < 1000000000)
+        return 9;
     return 10;
 }
 
+/**
+ * @brief Get the number of digits closest to it's "odd" value
+ *
+ * Useful for centering text on tiles
+ *
+ * @param n value to find the number of odd decimal digits of
+ *
+ * @return The closest number of digits that is an odd value.
+ */
 static inline int get_digits_odd(int n)
 {
-    if (n < 100) return 1;
-    if (n < 10000) return 2;
-    if (n < 1000000) return 3;
-    if (n < 100000000) return 4;
+    if (n < 100)
+        return 1;
+    if (n < 10000)
+        return 2;
+    if (n < 1000000)
+        return 3;
+    if (n < 100000000)
+        return 4;
     return 5;
 }
 
+/**
+ * @brief Get the number of digits closest to it's "even" value
+ *
+ * Useful for centering text on tiles
+ *
+ * @param n value to find the number of even decimal digits of
+ *
+ * @return The closest number of digits that is an even value.
+ */
 static inline int get_digits_even(int n)
```

Comment:
```suggestion
/**
 * @brief Get the number of digits closest to it's "even" value, i.e. ceil(digits/2)
 *
 * Useful for centering text on tiles
 *
 * @param n value to find the number of even decimal digits of
 *
 * @return The value of ceil(get_digits(n)/2)
 */
static inline int get_digits_even(int n)
```

A more accurate documentation. Not bothering for `get_digits_odd()` because it's removed in #269.